docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,1385 @@
+- Dependencies MUST be injected via constructor.
+  - Models MUST NOT instantiate any object but [Transfer Objects](#transfer-object).
+- Dependencies MUST be referred by interface.
+- Dependencies MUST be from the same module, either `models`, [Repository](#repository), [Entity Manager](#entity-manager), [Config](#module-configuration), or [Bridge wrapped facade](#bridge).
+  - Models MUST NOT interact directly with other module's models (eg: via inheritance, shared constants, instantiation, etc.).
+- Models MUST be grouped under a folder by activity (eg: `Writers/ProductWriter.php`, `Readers/ProductReader.php`).
+- Models MUST NOT be named using generic words, such as `Executor`, `Handler`, or `Worker`.
+- Model methods MUST NOT be `private`.
+
+<details><summary markdown='span'>Additional Conventions for Project Development</summary>
+- Dependencies can be referred directly via class.
+</details>
+
+## Module Configuration
+
+**Description**
+- **Module configuration**: module specific, environment independent configuration in `[Module]Config.php.`
+- **Environment configuration**: configuration keys in `[Module]Constants.php` that can be controlled per environment via `config_default.php`.
+
+The module configuration class can access the environment configuration values via `$this->get('key')`.
+The module configurations are strictly defined in `Shared` application layer only, yet can be accessed from any application layer using the layer specific `Config`.
+
+**Conventions**
+
+- Module configuration
+  - Getter methods (prefixed with `get`) MUST be defined to retrieve configuration values.
+    - `Protected` constants MAY also be defined to support the getter methods (example: to enable cross-module referencing via `@uses`).
+- Environment configuration
+  - constants MUST be always `public` and have specification about their purpose.
+  - constants MUST contain the same UPPERCASE value as the key is + properly prefixed with MODULE_NAME (see Example below).
+
+<details><summary markdown='span'>Additional Conventions for Boilerplate, Accelerator, and Module Development</summary>
+- The module configuration relays exclusively on `static::` to support extension.
+</details>
+
+<details><summary markdown='span'>Guidelines for Module Development</summary>
+- `Protected` items are NOT recommended in module configuration as they tend to create BC problems on project extensions.
+</details>
+
+**Example**
+
+```php
+namespace Spryker\Shared\Oms;
+
+interface OmsConstants
+{
+    /**
+     * Specification:
+     * - Defines if OMS transition log is enabled.
+     *
+     * @api
+     *
+     * @var string
+     */
+    public const ENABLE_OMS_TRANSITION_LOG = 'OMS:ENABLE_OMS_TRANSITION_LOG';
+}
+```
+
+
+## Navigation.XML
+
+**Description**
+
+Represents the module entries of the Backoffice navigation panel.
+The icons are taken from [Font Awesome Icons Library](#https://fontawesome.com/v4/icon/files-o).
```

Comment:
Where is the example of how to use them?

Reply:
i've deleted by mistake, thank you!